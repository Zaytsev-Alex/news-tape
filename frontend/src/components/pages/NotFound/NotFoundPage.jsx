import React from 'react';
import {Card} from 'react-rainbow-components';
import {Link} from 'react-router-dom';
import PageWrapper from '../../common/PageWrapper';
import {ROOT} from '../../../constants/routerPaths';
import './index.css';

const NotFoundPage = () => {
    return (
        <PageWrapper className="content-center" hideNavigation>
            <Card className="not-found">
                <div>
                    <h1>Page not found</h1>
                    <p>Sorry. We couldn't find the page you're looking for.</p>
                </div>
                <Link to={ROOT} className="not-found__link link">
                    Go to home page
                </Link>
            </Card>
        </PageWrapper>
    );
};

export default NotFoundPage;
