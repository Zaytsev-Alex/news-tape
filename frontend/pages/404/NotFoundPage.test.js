import NotFoundPage from './index';
import {act, render, screen} from '@testing-library/react';
import {BrowserRouter} from 'react-router-dom';

describe('NotFoundPage tests', () => {
    let container = null;

    beforeEach(() => {
        act(() => {
            const renderRes = render(
                <BrowserRouter>
                    <NotFoundPage/>
                </BrowserRouter>
            );
            container = renderRes.container;
        });
    });

    it('NotFoundPage should be rendered', () => {
        expect(screen.getByText('Page not found')).toBeInTheDocument();
    });

    it('Go to home page should be rendered on the NotFoundPage', () => {
        expect(screen.getByText('Go to home page')).toBeInTheDocument();
    });

    it('NotFoundPage should match snapshot', () => {
        expect(container).toMatchSnapshot();
    });
});
