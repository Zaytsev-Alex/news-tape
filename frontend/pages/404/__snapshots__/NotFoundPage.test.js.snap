// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotFoundPage tests NotFoundPage should match snapshot 1`] = `
<div>
  <section
    class="app content-center"
  >
    <header
      class="header"
    >
      <a
        class="logo-link"
        href="/"
      >
        <img
          alt="go to home page"
          class="logo-link__img"
          src="logo.svg"
        />
      </a>
    </header>
    <main
      class="main"
    >
      <article
        class="sc-btzYZH dSiMTE not-found not-found"
      >
        <div>
          <h1>
            Page not found
          </h1>
          <p>
            Sorry. We couldn't find the page you're looking for.
          </p>
        </div>
        <a
          class="not-found__link link"
          href="/"
        >
          Go to home page
        </a>
      </article>
    </main>
    <footer
      class="footer"
    >
      Â© 2022 Copyright News Tape
    </footer>
  </section>
</div>
`;
